Makefile

```diff
@@ -489,6 +489,19 @@ endif
 CFLAGS += $(C_WARNING_FLAGS) $(WARNING_FLAGS) -I. -I./include $(PLATFORM_CCFLAGS) $(OPT)
 CXXFLAGS += $(WARNING_FLAGS) -I. -I./include $(PLATFORM_CXXFLAGS) $(OPT) -Woverloaded-virtual -Wnon-virtual-dtor -Wno-missing-field-initializers
 
+TARGETS_NO_BUILD=tags tags0 clean format check-format check-buck-targets
```

Comment:
watch-log, uninstall, install-headers...

Reply:
Will add more to the list.

Reply:
Is it common to provide invalid options with non-build targets and require the target to succeed? This feels like a detour.

Reply:
On second thought, maybe it is common, since people likely provide -DNIOSTATS_CONTEXT through EXTRA_CXXFLAGS which wouldn't be provided to `make clean` etc. Do we have a standard way of detecting whether compilation is not going to happen at all?

Reply:
@ajkr, I am not aware of a way to detect whether compilation will happen. However, I am not a Makefile expert, and would love to learn if there is a way of doing it.

Reply:
@riversand963 @ajkr There is an attempt in the Makefile to determine when compilation is not going to occur.  The rules around generating %.d (ROCKSD_DEP_RULES) do this check.  But some of the checks here are still compilation.  But it might be a good start...